slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/interchange/AttestationEpochManager.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.web3signer.slashingprotection.interchange;
+
+import tech.pegasys.web3signer.slashingprotection.dao.LowWatermarkDao;
+import tech.pegasys.web3signer.slashingprotection.dao.SignedAttestationsDao;
+import tech.pegasys.web3signer.slashingprotection.dao.SigningWatermark;
+import tech.pegasys.web3signer.slashingprotection.dao.Validator;
+
+import java.util.Optional;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tuweni.units.bigints.UInt64;
+import org.jdbi.v3.core.Handle;
+
+public class AttestationEpochManager {
+
+  private static final Logger LOG = LogManager.getLogger();
+
+  private final LowWatermarkDao lowWatermarkDao;
+
+  private final Optional<UInt64> highestSourceEpoch;
+  private final Optional<UInt64> highestTargetEpoch;
+  private final Handle h;
+  private final Validator validator;
+
+  final OptionalMinValueTracker minSourceTracker = new OptionalMinValueTracker();
+  final OptionalMinValueTracker minTargetTracker = new OptionalMinValueTracker();
+
+  private AttestationEpochManager(
```

Comment:
nit: AttestationWatermarkEpochManager? still not great but makes it clearer this is only about managing the watermarks not the attestation epochs

Reply:
renamed to AttestationWatermarkUpdater